modules/statistics/CallStats.js

```diff
@@ -65,167 +61,483 @@ const reportType = {
     MST_WITH_USERID: 'mstWithUserID'
 };
 
-/* eslint-enable no-invalid-this */
-
 /**
- * Returns a function which invokes f in a try/catch block, logs any exception
- * to the console, and then swallows it.
+ * An instance of this class is a wrapper for the CallStats API fabric. A fabric
+ * reports one peer connection the the CallStats backend and is allocated with
+ * {@link callstats.addNewFabric}. It has a bunch of instance methods for
+ * reporting various events. A fabric is considered disposed when
+ * {@link CallStats.sendTerminateEvent} is executed.
  *
- * @param f the function to invoke in a try/catch block
- * @return a function which invokes f in a try/catch block, logs any exception
- * to the console, and then swallows it
+ * Currently only one backend instance can be created ever and it's done using
+ * {@link CallStats.initBackend}. At the time of this writing there is no way to
+ * explicitly shutdown the backend, but it's supposed to close it's connection
+ * automatically, after all fabrics have been terminated.
```

Comment:
I think I might have seen this fail, as in, stuff still happening when the call ends on mobile. Any chance we can forcefully terminate it?